iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -81,10 +107,13 @@ inline void function_ref<ReturnType(ArgTypes...)>::swap(function_ref<ReturnType(
 }
 
 template <class ReturnType, class... ArgTypes>
+// AXIVION Next Line AutosarC++19_03-A2.10.4 : Overload for swap(function_ref, function_ref) as in STL
```

Comment:
Why is this needed at all? If this rule is about not using STL names it is problematic for us as we reimplement some of the STL functionality and keep the function names like in STL where possible.

Reply:
It is probably not a good rule for us to activate then ... but also not a question for this PR